File: server/app/providers/adapters/ollama.py
from __future__ import annotations

from langchain_ollama import ChatOllama

from ..base import ProviderRuntimeConfig


class OllamaAdapter:
    provider_type = "ollama"

    def validate(self, config: ProviderRuntimeConfig) -> None:
        if config.model.strip() == "":
            raise ValueError("ollama provider requires model")

    def create_model(self, config: ProviderRuntimeConfig) -> ChatOllama:
        self.validate(config)
        kwargs = {
            "model": config.model,
            "base_url": config.base_url or "http://127.0.0.1:11434",
        }
        if config.temperature is not None:
            kwargs["temperature"] = config.temperature
        return ChatOllama(**kwargs)

File: desktop/src/runtime/providers/adapters/ollama.ts
import { ChatOllama } from "@langchain/ollama";
import type { ProviderType } from "@ai-emotion/contracts";
import type { ProviderAdapter, ProviderRuntimeConfig } from "../provider-service";

export class OllamaAdapter implements ProviderAdapter {
  public readonly providerType: ProviderType = "ollama";

  public validate(config: ProviderRuntimeConfig): void {
    if (config.model.trim() === "") {
      throw new Error("ollama provider requires model");
    }
  }

  public createModel(config: ProviderRuntimeConfig): ChatOllama {
    this.validate(config);

    return new ChatOllama({
      model: config.model,
      baseUrl: config.base_url ?? "http://127.0.0.1:11434",
      temperature: config.temperature ?? undefined
    });
  }
}
